cranelift-wasm/src/code_translator.rs

```diff
@@ -1028,6 +1028,22 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
             let (a, b) = state.pop2();
             state.push1(builder.ins().imul(a, b))
         }
+        Operator::V128Not => {
+            let a = state.pop1();
+            state.push1(builder.ins().bnot(a));
+        }
+        Operator::V128And => {
+            let (a, b) = state.pop2();
+            state.push1(builder.ins().band(a, b));
+        }
+        Operator::V128Or => {
+            let (a, b) = state.pop2();
+            state.push1(builder.ins().bor(a, b));
+        }
+        Operator::V128Xor => {
```

Comment:
Light suggestion: it seems we could commonize all this code with the non-vector branches too, which would result in less code, but would break the grouping of SIMD opcodes. I leave it to your personal taste.

Reply:
Opened https://github.com/CraneStation/cranelift/issues/1139.